### ISSUE TITLE
Doesn't seem to loook at my notes

### ISSUE THREAD

**Issue by gramster** [brianpetro/obsidian-smart-connections#495]
I am starting every query with "Based on my notes...", but every response starts with something like:

"I'm sorry for any confusion, but as an AI developed by OpenAI, I don't have the capability to access or remember personal notes or any specific details about users' previous interactions or inquiries. Each session with me is independent, ensuring privacy and confidentiality for users."

and goes on to be something generic.

**Comment by brianpetro** [comment]
Hey @gramster 

Can you send me some screenshots of the developer console and your settings?

Check out this issue for examples https://github.com/brianpetro/obsidian-smart-connections/issues/483#issue-2157757457

Thanks for your help in solving this
🌴

**Comment by brianlaughlin** [comment]
Got the same issue.
![img1](https://github.com/brianpetro/obsidian-smart-connections/assets/2857806/80010d76-1048-4dee-8ecb-65c13f4fd713)

version 2.0.122

**Comment by brianlaughlin** [comment]
Here are my settings
![img2](https://github.com/brianpetro/obsidian-smart-connections/assets/2857806/50bc6430-791c-48e6-9d8f-a960c6a1403b)

**Comment by brianpetro** [comment]
Thanks @brianlaughlin 

Can you also check the developer console?

🌴

**Comment by gramster** [comment]
Embeddings settings: 
<img width="793" alt="image" src="https://github.com/brianpetro/obsidian-smart-connections/assets/2762632/9b951dfe-b55b-4939-a270-4aba3779bc4d">

I have an OpenAI key and have selected gpt-4-turbo (128k); everything else is set to default.

If I click on 'Exclude all top level folders', I get this popup,; perhaps it is a clue 
<img width="208" alt="image" src="https://github.com/brianpetro/obsidian-smart-connections/assets/2762632/a1f1e340-3e1d-4b36-97bb-39451d332d54">

After clicking on that I remove my top-level folder from the excluded folders settings, so I am left with ".git, .github, .obsidian, .smart-connections, .trash"

I cleared deveeloper console and ran a query and then this is what I see:

<img width="526" alt="image" src="https://github.com/brianpetro/obsidian-smart-connections/assets/2762632/1570c0d9-4319-4a74-9913-cde5c9902542">

If I quit and restart, I first see:

<img width="523" alt="image" src="https://github.com/brianpetro/obsidian-smart-connections/assets/2762632/1862a841-82cd-449d-953d-19a1356be15e">

I get a prompt asking whether I want to embed some 400 documents. After proceeding, I see a boatload of:

<img width="521" alt="image" src="https://github.com/brianpetro/obsidian-smart-connections/assets/2762632/2295bf7d-cfc6-4ac4-9c90-103fd3111222">

Before that I saw a lot of:

<img width="512" alt="image" src="https://github.com/brianpetro/obsidian-smart-connections/assets/2762632/75371427-760b-45ba-bb69-98ba77201b27">

**Comment by brianpetro** [comment]
This screenshot indicates things are working 
<img width="526" alt="image" src="https://github.com/brianpetro/obsidian-smart-connections/assets/1886014/385b761c-b297-4e80-9548-e8ad4ddbe9b4">

What was the response from the chat to the query you ran?

The errors in the other screenshots are OK.

Lastly, I have issues running the Jina 8K model on my M2 Mac 8GB, so it might be worth trying with a different model. I recommend using the BGE-micro model for both notes/blocks until this gets solved.

**Comment by brianlaughlin** [comment]
> Thanks @brianlaughlin
> 
> Can you also check the developer console?
> 
> 🌴
Here you go. I am happy to run other tests. I'll restart Obsidian and do the same. 

![img3](https://github.com/brianpetro/obsidian-smart-connections/assets/2857806/f0738bcb-69cd-4fce-861f-ae694c4f0cf1)

**Comment by brianlaughlin** [comment]
This is onload.
![onload](https://github.com/brianpetro/obsidian-smart-connections/assets/2857806/deea24dc-9af8-40f0-9dbc-f2694d5b6908)

Version was auto updated to:
Looks like it went back on version, ran test still failed. Updated back to 2.0.122.

**Comment by brianpetro** [comment]
Ok, now we're getting somewhere.

It looks like it's failing to retrieve relevant notes.

To help improve the value of the logs, please turn on this setting:

![Screenshot 2024-02-27 at 6 13 25 PM](https://github.com/brianpetro/obsidian-smart-connections/assets/1886014/dd068a7c-4a7b-4226-98c9-e832b72493ae)

Is the Smart View working?

If it isn't working, this would be a good place to start.

The log line that says "bad escaped character" is probably indicative of what's causing the issues, and this could be causing additional issues, like re-embedding every time you turn on Obsidian/Smart Connections.

A "force refresh" (in the settings) might fix things.

So try toggling on that setting, then restart Obsidian, try a force refresh, observe the logs, try the chat again, and let me know how it goes.

Thanks for your help in solving this!
🌴

**Comment by brianlaughlin** [comment]
Of course. Yes, I have had several forced refreshes, but I figured you're just working on something. :-) I'll send you what I find out. I'm happy to help.

**Comment by gramster** [comment]
@brianpetro I have 32GB RAM; do you still think I should change embeddings library?

**Comment by brianlaughlin** [comment]
Did the following:
1. Enabled Debug at Start up.
2. Restarted
3. Did a force refresh
4. When completed ran test again.
5. Test failed.
Here is the log
![log](https://github.com/brianpetro/obsidian-smart-connections/assets/2857806/3d420bcb-d0f6-42e7-8dde-3225d89804e2)

Smart connections does seem to work on the panel, well it at least shows results. But if I type a ```smart-connections in a note that doesn't work anymore.

**Comment by gramster** [comment]
Trying with that micro embeddings model. I get a lot of:

<img width="497" alt="image" src="https://github.com/brianpetro/obsidian-smart-connections/assets/2762632/d4e2938f-f9ad-4d77-83f2-6024729a5bda">

I still saw this at the end:

<img width="526" alt="image" src="https://github.com/brianpetro/obsidian-smart-connections/assets/2762632/c1027e9a-bb0a-4bda-a8e1-f38564c675a6">

And I still have the same problem that my notes are not getting consulted. So it is not the embeddings model.

**Comment by brianpetro** [comment]
@brianlaughlin I'm pretty certain that "Bad escaped character" error is the culprit. Are there any files with slashes or quotes in the name? While it shouldn't, they may be causing the issue.

@gramster that should be plenty of RAM for Jina-8k. The truncating notices are expected for BGE-micro. The error suggests that the Note-level embedding is missing for some reason.

It might make sense for both of you to temporarily turn off the block-level embeddings (set to None). Then see how things work after that.

If it's still not working, similarly, set the Note-level embeddings to None and then only use block-level embeddings to see if that works.

This way, we might be able to narrow down the source of the issue.

Thanks again for helping me figure this out!

🌴

**Comment by brianpetro** [comment]
@brianlaughlin that is very clever; nice work!

Did you disable block-level embeddings (set to "None") and see if the errors still occur?

It could be that the problematic characters occur in headings within notes, and that's why it's the block-level embeddings file that's failing to load.

**Comment by brianlaughlin** [comment]
@brianpetro 
That kinda solved it.
Steps
1. Disabled
2. It redid the embeddings.
3. Restarted - new version found. Upgraded to 2.1
4. Restarted.

Now it's talking to notes again. That's the good part.
It failed the test to find the phone number. That is one of the reasons I decided to use OpenAI's embeddings because it would return better quality results when I tested it against others. 

So this problem is resolved. Feel free to ask if there is anything I can do to help improve processes. Thank you for your amazing dedication to a wonderful tool.

**Comment by brianpetro** [comment]
@brianlaughlin  😊

Thanks for your help in getting to the bottom of the issue!

🌴

**Comment by gramster** [comment]
I turned off block embeddings too and things are working now.

**Comment by brianpetro** [comment]
@gramster a step in the right direction! 

Thanks for following up
🌴

**Comment by xinyu-dev** [comment]
Same issue here. Like @gramster  I also have to turn off block embedding in order for it to work again.

**Comment by jayeshmasand03** [comment]
@brianpetro another issue I noticed. 2 of my notes (Daily and weekly review templates) were not accessible by Obsidian because it is owned by another user (another user in mac). This was resulting in embedding getting stuck at 0. 

Added Templates folder in the exclusion and it is working like a charm. I thought I should let you know about the same
